src/http_server.cc

```diff
@@ -1353,8 +1354,24 @@ HTTPAPIServer::HandleRepositoryControl(
             HTTP_RESPOND_IF_ERR(
                 req,
                 param_json.MemberAsString(m.c_str(), &param_str, &param_len));
-            auto param = TRITONSERVER_ParameterNew(
-                m.c_str(), TRITONSERVER_PARAMETER_STRING, param_str);
+
+            TRITONSERVER_Parameter* param = nullptr;
+            if (m == "config") {
+              param = TRITONSERVER_ParameterNew(
+                  m.c_str(), TRITONSERVER_PARAMETER_STRING, param_str);
+            } else if (m.rfind("file:", 0) == 0) {
+              // Decode base64
+              base64_decodestate s;
+              base64_init_decodestate(&s);
+
+              // The decoded can not be larger than the input...
+              binary_files.emplace_back(std::vector<char>(param_len + 1));
+              size_t decoded_size = base64_decode_block(
+                  param_str, param_len, binary_files.back().data(), &s);
```

Comment:
Should this still call `base64_decode_block` ? I noticed in `lifecycle_test.py` changes you removed the b64 encoding :
```python
# before
encoded_file = base64.b64encode(f.read())
# after
file_content = f.read()
```

But I see the client library is doing some B64 encoding, so maybe that's OK

Reply:
For HTTP the encoding is still required to transfer via JSON, the client side changes now handles the encoding implicitly for ease of use. If you are forming HTTP request outside the client library, the encoding will need to be added.

Reply:
> If you are forming HTTP request outside the client library, the encoding will need to be added.

Is this documented anywhere, even if just a code comment? IIRC you mentioned this was an internal feature, not expected for public users.

Reply:
It is documented in the [protocol](https://github.com/triton-inference-server/server/blob/main/docs/protocol/extension_model_repository.md#load) that file content sent via HTTP load API must be base64 encoded, so the users should follow that if they are forming the HTTP on their own. Client library is wrapper over the protocol and it will take care of the encoding for the user.